sdk/core/azure-core/inc/azure/core/internal/io/filehandleholder.hpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief An internal FILE* Holder.
+ *
+ */
+#pragma once
+
+#include <cstdio>
+
+namespace Azure { namespace IO { namespace Internal {
+
+  /**
+   * @brief FileHandleHolder ensures native handle resource is released.
+   */
+  class FileHandleHolder {
+  private:
+    FILE* m_value = nullptr;
+
+  public:
+    explicit FileHandleHolder() noexcept : m_value(nullptr){};
+
+    ~FileHandleHolder() noexcept
+    {
+      if (m_value != nullptr)
+      {
+        fclose(m_value);
+        m_value = nullptr;
+      }
+    }
+
+    void operator=(FILE* p) noexcept
```

Comment:
Add comments about this class is expected to be init with this operator.  (But that's unusual,.. use a cttor better)